### PR TITLE
add php 5.6 and php 7

### PR DESCRIPTION
**Pull request by omansour**

### CODE REVIEW

**Comment by omansour** [comment]
waiting for https://github.com/pdezwart/php-amqp/pull/173

**Comment by pinepain** [comment]
Pull https://github.com/pdezwart/php-amqp/pull/181 merged and beta released on PECL (https://pecl.php.net/package-changelog.php?package=amqp&release=1.7.0alpha1)

**Comment by omansour** [comment]
that's so cool ! thanks @pinepain :)

**Comment by omansour** [comment]
@pinepain I have some troubles to setup the alpha2 version with travis. any idea ? https://travis-ci.org/M6Web/AmqpBundle/jobs/98194643 

txs a lot

**Comment by pinepain** [comment]
@omansour, looks like librabbitmq not found, on [line 180](https://travis-ci.org/M6Web/AmqpBundle/jobs/98194643#L180): 
`configure: error: Please reinstall the librabbitmq distribution itself or (re)install librabbitmq development package if it available in your system`

Look how into [.travis.yml:56](https://github.com/pdezwart/php-amqp/blob/f939154d2fbf6e22b1111ebb5e1e9a0abfb6fa14/.travis.yml#L56) from php-amqp extension:
`sh provision/install_rabbitmq-c.sh ${LIBRABBITMQ_VERSION}`

Feels free to take `provision/install_rabbitmq-c.sh` (just add some copyright) or install librabbitmq from repo if it has fresh version. Since php-amqp>=1.6.0 librabbitmq >= 0.5.2 required, >= 0.6.0 recommended. Starting from ubuntu 15.05 (vivid) 0.5.2 or newer included in repo, but even with [The Trusty beta Build Environment](https://docs.travis-ci.com/user/trusty-ci-environment/) you will have [0.4.1](http://packages.ubuntu.com/source/trusty/librabbitmq), so it looks like you have to run into full building cycle.

In case you decide to compile librabbitmq from sources (and it looks like you'll have to), note, that you may have to explicitly provide path to librabbitmq when compile php-amqp extension via `--with-librabbitmq-dir[=DIR]` option which `Set the path to librabbitmq install prefix.` in case you will have librabbitmq installed not somewhere in standard system paths (this is exactly the case for container-based builds, see below)

Also pay attention to [Using container-based infrastructure](https://docs.travis-ci.com/user/workers/container-based-infrastructure), it really speedups builds, don't know how this applied to [their move to GCE](https://blog.travis-ci.com/2015-11-27-moving-to-a-more-elastic-future), if you will be getting into the details, please, share what you will find out.

**Comment by omansour** [comment]
txs @pinepain. I manage to build the full things using travis - but I am note sure. There is some coredumps launched during the units tests https://travis-ci.org/M6Web/AmqpBundle/jobs/98293980 

strange thing is that using alpha1 packaged by Remi and found at RemiRepo, no problems are found. Surely something wrong on my side ...

**Comment by pinepain** [comment]
Segfaults are not what expected by any means. Is any segfault happens when you run tests locally?

**Comment by omansour** [comment]
same tests are failing locally ... 

```
[SU__________________________________________________________][2/2]
=> Test duration: 0.00 second.
=> Memory usage: 0.00 Mb.
> M6Web\Bundle\AmqpBundle\Tests\Units\Factory\ProducerFactory...
[SU__________________________________________________________][2/2]
=> Test duration: 0.00 second.
=> Memory usage: 0.00 Mb.
> Total tests duration: 0.15 second.
> Total tests memory usage: 0.00 Mb.
> Running duration: 5.41 seconds.
Failure (4 tests, 16/18 methods, 0 void method, 0 skipped method, 2 uncompleted methods, 0 failure, 0 error, 0 exception)!
> There are 2 uncompleted methods:
=> M6Web\Bundle\AmqpBundle\Tests\Units\Factory\ConsumerFactory::testFactory() with exit code -1:
==> output(0) ""
=> M6Web\Bundle\AmqpBundle\Tests\Units\Factory\ProducerFactory::testFactory() with exit code -1:
==> output(0) ""
```

I will try to get more informations on where the test fail. Maybe its not related to the extension.

**Comment by pinepain** [comment]
I'm also looking into the problem, it looks like the problem in mock part.

**Comment by pinepain** [comment]
Should be fixed in PR https://github.com/pdezwart/php-amqp/pull/194. See commit that addresses  segfault fixing when channel zval type != IS_OBJECT, I guess there will be no more rebasing, so it should be this commit - https://github.com/pinepain/php-amqp/commit/41a47dad05508da1cfa93ceb86b5d3216f11351a.

File: travis/5.5.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+yes | pecl install -f amqp-1.6.1
```
Comment (pinepain) [inline travis/5.5.sh]: With php-amqp >= 1.7.0 both PHP 7 and PHP 5 supported, so feels free to install 1.7.0. It is even recommended for the sake of API consistency while 1.7.0 introduces new API methods and extend or cleanup existent one.

File: travis/5.6.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+yes | pecl install -f amqp-1.6.1
```
Comment (pinepain) [inline travis/5.6.sh]: Same here, php-amqp support PHP >= 5.3 (before 1.7.0 even 5.2 was known to be compatible, though we removed it from build matrix as too outdated).

**Comment by omansour** [comment]
yes clearly ! core is dumped while testing if the Producer is an object
I will test asap

**Comment by pinepain** [comment]
I also provided some notices on commits, the idea is that starting from php-amqp 1.7.0 both PHP 7 and PHP 5 supported so no need to install older 1.6.x for PHP 5

**Comment by omansour** [comment]
@pinepain using your fork I have others errors :) https://travis-ci.org/M6Web/AmqpBundle/jobs/98477686 

sounds related to the mock too

**Comment by pinepain** [comment]
This is a userland problem. You are trying to perform channel operation without providing channel. A saw your stubs (or mocks), you have to rather stub or mock all methods which uses underlying channel or use real connection and channel. Check that commit I referenced above and look into the tests, you'll see the scenario for the exact case you experienced in your test.

**Comment by omansour** [comment]
@pinepain thanks a lot for your help ! The 1.7.0 behavior is much more safe as its throw an exception when you want to declare incorrectly a queue or an exchange. 

the ci is passing using your fork and the alpha2. Thanks again and merry Christmas to you.

**Comment by pinepain** [comment]
You welcome, glad to help. I guess after Christmas we will release a new version on pecl so it would be simpler to install. Merry Christmas! :christmas_tree:

**Pull request merged**